### PR TITLE
Lims 3680

### PR DESCRIPTION
**Pull request by ghada-37**

### CODE REVIEW

File: ui_testing/testcases/basic_tests/test04_testunits.py [hunk 1]
```diff
@@ -777,169 +777,71 @@ def test023_duplicate_test_case(self):
...
         self.assertGreater(new_test_units, old_test_units)
 
-    @parameterized.expand(['unitsub', 'unitsuper'])
-    def test029_qualitative_test_unit_with_sub_and_super_scripts_appears_in_exported_sheet(self,
-                                                                                           unit_with_sub_or_super):
+    @parameterized.expand([('unitsub', 'qualitative'),
+                           ('unitsub', 'quantitative'),
+                           ('unitsuper', 'qualitative'),
+                           ('unitsuper', 'quantitative')])
+    def test023_test_unit_with_sub_and_super_scripts_appears_in_exported_sheet(self, unit_with_sub_or_super, type):
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test04_testunits.py]: Run those test cases in travis

File: ui_testing/testcases/basic_tests/test04_testunits.py [hunk 2]
```diff
@@ -1017,36 +919,29 @@ def test025_edit_approach_overview_button(self):
...
-            self.assertTrue(self.test_unit_page.check_for_qualitative_fields())
-        elif testunit_type == 'Quantitative MiBi':
-            self.assertTrue(self.test_unit_page.check_for_quantitative_mibi_fields())
-
-    def test_029_allow_user_to_change_between_specification_and_quantification(self):
+        self.info('edit random testunit')
+        self.test_unit_page.get_random_test_units()
+        self.base_selenium.get_url()
+        testunit_types = ['Quantitative', 'Qualitative', 'Quantitative MiBi']
+        for testunit_type in testunit_types:
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test04_testunits.py]: This is not correct, For loop doesn't equal the parameterized. For loop means it is only one test case, parameterized means there r n test cases.

File: ui_testing/testcases/basic_tests/test04_testunits.py (hunk 1 above)
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test04_testunits.py]: This test case isn't related to this branch, so I don't need to run them in travis

File: ui_testing/testcases/basic_tests/test04_testunits.py (hunk 2 above)
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test04_testunits.py]: it's already one test case : 
https://github.com/modeso-open-source/1lims-automation/blob/LIMS-3680/ui_testing/testcases/basic_tests/test04_testunits.py#L922

Comment (0xIslamTaha) [reply ui_testing/testcases/basic_tests/test04_testunits.py]: no, please separate them those are 3 different test case
as general rulse:

1. - Test case should test only one case
2. - Test case should have as minimum assertion as possible

Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test04_testunits.py]: ok, it's a good point

**Pull request closed without merging**